jdaviz/configs/cubeviz/helper.py

```diff
@@ -126,3 +130,57 @@ class CubeViz(Cubeviz):
 
 def layer_is_cube_image_data(layer):
     return isinstance(layer, BaseData) and layer.ndim in (2, 3)
+
+
+# TODO: We can remove this when specutils supports it, i.e.,
+#       https://github.com/astropy/specutils/issues/592 and
+#       https://github.com/astropy/specutils/pull/1009
+# NOTE: Cannot use custom_write decorator from specutils because
+#       that involves asking user to manually put something in
+#       their ~/.specutils directory.
+
+def jdaviz_cube_fitswriter(spectrum, file_name, **kwargs):
+    """This is a custom writer for Spectrum1D data cube.
+    This writer is specifically targetting data cube
+    generated from Cubeviz plugins (e.g., cube fitting)
+    with FITS WCS. It writes out data in the following format
+    (with MASK only exist when applicable)::
+
+        No.    Name      Ver    Type
+          0  PRIMARY       1 PrimaryHDU
+          1  SCI           1 ImageHDU (float32)
+          2  MASK          1 ImageHDU (uint16)
```

Comment:
If this is the file structure you need, I don't think `wcs1d-fits` can do it in its present form – with or without the new PR. You can write to the first extension instead of the primary with `write(..., hdu=1, format='wcs1d-fits'), but currently there is no provision to write the mask. It could certainly be added, but last I recall specutils had rather poor support to write to a new HDU in an existing HDUList.
If this loader is working as it should, might be best to use it as is and perhaps then contribute to the `default_loaders` upstream. Would this make sense to provide as a writer for `jwst_s3d_loader`, or is that yet another format?

Reply:
This has nothing to do with JWST, it is a "generic" format for a cube output from model fitting plugin that could or could not have mask populated. So putting this in `jwst_s3d_loader` would be very misleading.

Reply:
In that case it would probably belong into `wcs1d-fits`. I have not found much on mask treatment apart from MaNGA/SDSS, so don't know what would be a standard for storing it.

Reply:
I dunno either. If this is not generic enough, personally, I am okay with this writer just living in Jdaviz forever.